### PR TITLE
(feat): Allow hiding card title in custom homepage cards

### PR DESCRIPTION
**Pull request by ivangonzalezacuna**
## Hey, I just made a Pull Request!

In some cases, the Card header is not really important, since it just occupies space that might be used by other things. With this change both the custom cards that a person creates as well as the built-in ones (or others provided by external plugins) can hide that text.

There are 2 options to do it:

1. Inside the `CustomHomePageGrid`:
    ```ts
      <CustomHomepageGrid>
        <HomePageStarredEntities hideTitle />
      </CustomHomepageGrid>
    ```
2. As part of the `createCardExtension`:
    ```ts
    export const TestHomePageComponent = homePlugin.provide(
      createCardExtension({
        name: 'Test',
        title: 'Test',
        hideTitle: true,
        components: () => import('./homePageComponents/Test'),
      }),
    );
    ```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/home-react/src/extensions.tsx [hunk 1]
```diff
@@ -47,7 +47,10 @@ export type RendererProps = { title: string } & ComponentParts;
 /**
  * @public
  */
-export type CardExtensionProps<T> = ComponentRenderer & { title?: string } & T;
+export type CardExtensionProps<T> = ComponentRenderer & {
+  title?: string;
```
Comment (camilaibs) [inline plugins/home-react/src/extensions.tsx:51]: Suggestion: Can the title prop and option be made optional? The default renderer could hide the card title if it isn't passed... It seems intuitive to me. How do you feel about the idea?

**Comment by camilaibs** [comment]
Starting to review 🙂

**Review by camilaibs — COMMENTED** [review]
Thanks so much for contributing 🎉 ! Commented with a minor suggestion.

File: plugins/home-react/src/extensions.tsx (hunk 1 above)
Comment (camilaibs) [reply plugins/home-react/src/extensions.tsx:51]: Sure, I mean in [this](https://github.com/backstage/backstage/blob/f545032b204f1fa5498cfea12ec14b2f1b28e27e/plugins/home-react/src/extensions.tsx#L161) line only add title when it is defined:

```tsx
// plugins/home-react/src/extensions.tsx

// In the <CardExtension/> component, line 161
const cardProps = {
...(title && { title })
}
```

Also, remove the title from the extended `CardExtensionComponentProps` type on [line 122](https://github.com/backstage/backstage/blob/f545032b204f1fa5498cfea12ec14b2f1b28e27e/plugins/home-react/src/extensions.tsx#L122C2-L122C2) and change to optional the title on the createCardExtension function options type, [line 85](https://github.com/backstage/backstage/blob/f545032b204f1fa5498cfea12ec14b2f1b28e27e/plugins/home-react/src/extensions.tsx#L85). 

Let me know if it clarifies the suggestion 🙌🏻

File: plugins/home-react/api-report.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ export type CardConfig = {
 // @public (undocumented)
 export type CardExtensionProps<T> = ComponentRenderer & {
   title?: string;
+  hideTitle?: boolean;
```
[the remark this answers is not in the document]
Comment (ivangonzalezacuna) [reply plugins/home-react/api-report.md]: Makes total sense. I tested it in our local instance and everything worked as expected. If you set a `name` but the `title` is left empty, then the title is hidden.

Also, I fixed the type issue.

**Review by camilaibs — APPROVED** [review]
That's fantastic ⭐ ! We would be happy to have you open a follow-up pull request adding tests for these components if you have time to do so and contribute with us again 🙂 .

**Comment by ivangonzalezacuna** [comment]
> That's fantastic star ! We would be happy to have you open a follow-up pull request adding tests for these components if you have time to do so and contribute with us again slightly_smiling_face .

Great! I could give it a try if I have the time. Testing in React and Typescript is a bit hard for me still, since I'm still in learning it, but I can check if some things can be added at least :)

**Pull request merged**